enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/IdentityVerificationCreateService.java

```diff
@@ -18,35 +18,65 @@
 import com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus;
 import com.wultra.app.enrollmentserver.model.integration.OwnerId;
+import com.wultra.security.powerauth.client.PowerAuthClient;
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import com.wultra.security.powerauth.client.v3.ListActivationFlagsResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Service implementing creating of identity verification.
  *
  * @author Lukas Lukovsky, lukas.lukovsky@wultra.com
+ * @author Roman Strobl, roman.strobl@wultra.com
  */
 @Service
 public class IdentityVerificationCreateService {
 
+    private static final Logger logger = LoggerFactory.getLogger(IdentityVerificationCreateService.class);
+
+    private static final String ACTIVATION_FLAG_VERIFICATION_PENDING = "VERIFICATION_PENDING";
+    private static final String ACTIVATION_FLAG_VERIFICATION_IN_PROGRESS = "VERIFICATION_IN_PROGRESS";
+
     private final IdentityVerificationRepository identityVerificationRepository;
+    private final PowerAuthClient powerAuthClient;
 
     /**
      * Service constructor.
      * @param identityVerificationRepository Identity verification repository.
+     * @param powerAuthClient PowerAuth client.
      */
     @Autowired
-    public IdentityVerificationCreateService(IdentityVerificationRepository identityVerificationRepository) {
+    public IdentityVerificationCreateService(IdentityVerificationRepository identityVerificationRepository, PowerAuthClient powerAuthClient) {
         this.identityVerificationRepository = identityVerificationRepository;
+        this.powerAuthClient = powerAuthClient;
     }
 
     @Transactional
-    public IdentityVerificationEntity createIdentityVerification(OwnerId ownerId) {
+    public IdentityVerificationEntity createIdentityVerification(OwnerId ownerId) throws DocumentVerificationException {
+        try {
+            ListActivationFlagsResponse response = powerAuthClient.listActivationFlags(ownerId.getActivationId());
+            List<String> activationFlags = new ArrayList<>(response.getActivationFlags());
+            if (!activationFlags.contains(ACTIVATION_FLAG_VERIFICATION_PENDING)) {
+                throw new DocumentVerificationException("Activation flag VERIFICATION_PENDING not found when initializing identity verification");
+            }
+            activationFlags.remove(ACTIVATION_FLAG_VERIFICATION_PENDING);
+            activationFlags.add(ACTIVATION_FLAG_VERIFICATION_IN_PROGRESS);
+            powerAuthClient.updateActivationFlags(ownerId.getActivationId(), activationFlags);
+        } catch (PowerAuthClientException ex) {
+            logger.error(ex.getMessage(), ex);
```

Comment:
It would be better to log something more than the raw log message which is logged in the stacktrace too.